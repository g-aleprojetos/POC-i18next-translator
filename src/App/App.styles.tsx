import styled from "styled-components";
import Imagei18NextSVG from "../assets/images/i18next.svg";

export interface IApp {
  "data-escolhido"?: boolean;
}

export const Button = styled.button<IApp>`
  width: 100px;
  height: 30px;
  margin: 10px;
  background-color: ${(props) => (props["data-escolhido"] ? "#aff" : "#aaa")};
`;


export const Container = styled.div`
  display: flex;
  flex-direction: column;
`;

export const ContainerArticle = styled.article`
  display: flex;
  padding: 10px 100px;
`;

export const ContainerButtons = styled.div``;

export const Header = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: center;
  width: 100%;
  height: 100px;
  background-color: #202024;
`;

export const Imagei18Next = styled.img.attrs({
  src: Imagei18NextSVG,
  alt: "i18Next",
})`
  width: 240px;
  height: 120px;
`;

export const Item = styled.a`
  margin: 10px;
`;

export const Nav = styled.nav`
  display: flex;
  margin-right: 100px;
`;

export const Main = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

export const TextDescription = styled.p`
  font-size: 20px;
  line-height: 1.8;
  color: #333;
  margin-bottom: 10px;
  text-align: justify;
`;

export const TextNav = styled.p`
  font-size: 15px;
  color: #fff;
  cursor: pointer;
`;
