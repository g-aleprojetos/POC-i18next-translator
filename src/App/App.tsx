import React from "react";
import { useTranslation } from "react-i18next";
import { namespaces } from "../utils/i18n";
import * as S from "./App.styles";

export const App = () => {
  const { t: header, i18n } = useTranslation(namespaces.pages.header);
  const { t: main } = useTranslation(namespaces.pages.main);

  return (
    <S.Container>
      <S.Header>
        <S.Nav>
          <S.Item>
            <S.TextNav>{header("home")}</S.TextNav>
          </S.Item>
          <S.Item>
            <S.TextNav>{header("contents")}</S.TextNav>
          </S.Item>
          <S.Item>
            <S.TextNav>{header("about")}</S.TextNav>
          </S.Item>
        </S.Nav>
      </S.Header>
      <S.Main>
        <S.ContainerButtons>
          <S.Button
            data-escolhido={i18n.resolvedLanguage === "pt"}
            type="submit"
            onClick={() => i18n.changeLanguage("pt")}
          >
            PortuguÃªs
          </S.Button>
          <S.Button
            data-escolhido={i18n.resolvedLanguage === "en"}
            type="submit"
            onClick={() => i18n.changeLanguage("en")}
          >
            English
          </S.Button>
          <S.Button
            data-escolhido={i18n.resolvedLanguage === "fr"}
            type="submit"
            onClick={() => i18n.changeLanguage("fr")}
          >
            FranÃ§ais
          </S.Button>
        </S.ContainerButtons>
        <S.Imagei18Next />
        <S.ContainerArticle>
          <S.TextDescription>{main("description")}</S.TextDescription>
        </S.ContainerArticle>
      </S.Main>
    </S.Container>
  );
};
