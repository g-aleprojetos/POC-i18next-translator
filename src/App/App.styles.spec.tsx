import React from "react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";
import * as S from "./App.styles";

describe("App.styles", () => {
  test("Button DEVE ser igual ao snapshot", () => {
    const button = renderer.create(<S.Button />).toJSON();
    expect(button).toMatchSnapshot();
  });

  test('Button DEVE ser igual ao snapshot QUANDO "escolhido" for igual a true', () => {
    const button = renderer.create(<S.Button data-escolhido={true} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  test("Container DEVE ser igual ao snapshot", () => {
    const container = renderer.create(<S.Container />).toJSON();
    expect(container).toMatchSnapshot();
  });

  test("ContainerArticle DEVE ser igual ao snapshot", () => {
    const containerArticle = renderer.create(<S.ContainerArticle />).toJSON();
    expect(containerArticle).toMatchSnapshot();
  });

  test("ContainerButtons DEVE ser igual ao snapshot", () => {
    const containerButtons = renderer.create(<S.ContainerButtons />).toJSON();
    expect(containerButtons).toMatchSnapshot();
  });

  test("Header DEVE ser igual ao snapshot", () => {
    const header = renderer.create(<S.Header />).toJSON();
    expect(header).toMatchSnapshot();
  });

  test("Imagei18Next DEVE ser igual ao snapshot", () => {
    const imagei18Next = renderer.create(<S.Imagei18Next />).toJSON();
    expect(imagei18Next).toMatchSnapshot();
  });

  test("Item DEVE ser igual ao snapshot", () => {
    const item = renderer.create(<S.Item />).toJSON();
    expect(item).toMatchSnapshot();
  });

  test("Nav DEVE ser igual ao snapshot", () => {
    const nav = renderer.create(<S.Nav />).toJSON();
    expect(nav).toMatchSnapshot();
  });

  test("Main DEVE ser igual ao snapshot", () => {
    const main = renderer.create(<S.Main />).toJSON();
    expect(main).toMatchSnapshot();
  });

  test("TextDescription DEVE ser igual ao snapshot", () => {
    const textDescription = renderer.create(<S.TextDescription />).toJSON();
    expect(textDescription).toMatchSnapshot();
  });

  test("Texto DEVE ser igual ao snapshot", () => {
    const texto = renderer.create(<S.TextNav />).toJSON();
    expect(texto).toMatchSnapshot();
  });
});
