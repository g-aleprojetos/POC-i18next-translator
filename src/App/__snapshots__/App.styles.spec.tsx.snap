// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.styles Button DEVE ser igual ao snapshot 1`] = `
<button
  className="gldefs"
/>
`;

exports[`App.styles Button DEVE ser igual ao snapshot QUANDO "escolhido" for igual a true 1`] = `
<button
  className="grjCIo"
  data-escolhido={true}
/>
`;

exports[`App.styles Container DEVE ser igual ao snapshot 1`] = `
<div
  className="dWlLnZ"
/>
`;

exports[`App.styles ContainerArticle DEVE ser igual ao snapshot 1`] = `
<article
  className="gRTedU"
/>
`;

exports[`App.styles ContainerButtons DEVE ser igual ao snapshot 1`] = `
<div
  className=""
/>
`;

exports[`App.styles Header DEVE ser igual ao snapshot 1`] = `
<div
  className="kVkDoM"
/>
`;

exports[`App.styles Imagei18Next DEVE ser igual ao snapshot 1`] = `
<img
  alt="i18Next"
  className="bEdBbZ"
  src="i18next.svg"
/>
`;

exports[`App.styles Item DEVE ser igual ao snapshot 1`] = `
<a
  className="jljOCQ"
/>
`;

exports[`App.styles Main DEVE ser igual ao snapshot 1`] = `
<div
  className="jBuTAN"
/>
`;

exports[`App.styles Nav DEVE ser igual ao snapshot 1`] = `
<nav
  className="gCQovH"
/>
`;

exports[`App.styles TextDescription DEVE ser igual ao snapshot 1`] = `
<p
  className="kMgaZE"
/>
`;

exports[`App.styles Texto DEVE ser igual ao snapshot 1`] = `
<p
  className="fwPkAE"
/>
`;
